lib/compiler/compileNative.ts

```diff
@@ -1,13 +1,13 @@
 import {join, resolve} from 'path';
 import {execSync} from 'child_process';
 import {Config} from '../config/config';
-import ImportMappingBuilder from './importMappingBuilder';
-import {buildInputObject, buildSources} from './buildUitls';
+import {buildInputObject, buildSourcesObject} from './buildUitls';
+import { ImportFile } from '@resolver-engine/imports';
 
 export function compileNative(config: Config) {
-  return async function compile(sources: string[]) {
+  return async function compile(sources: ImportFile[]) {
     const command = createBuildCommand(config);
-    const input = JSON.stringify(buildInputJson(sources, config), null, 2);
+    const input = JSON.stringify(buildInputObject(buildSourcesObject(sources)), null, 2);
```

Comment:
This could be simplified to just `buildInputObject(sources)` by having `buildInputObject` call `buildSourcesObject`